### PR TITLE
[#32] Implement character loader

### PR DESCRIPTION
**Pull request by xeoh**
Implement character loader(#32).

### 1. ```Hangul``` Enum class
```Hangul``` Enum represent Hangul UNICODE.

### 2. ```CharacterLoader```
Load ```HangulCharacter``` subclasses from ```Hangul```

### CODE REVIEW

**Review by bimak-server — APPROVED** [review]
👍

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  /**
+   * Get Hangul enum class from UNICODE value.
+   *
+   * @param value UNICODE value
+   * @return Correspond Hangul enum class
+   */
+  public static Hangul fromInt(int value) {
+    Alert.log(Hangul.class, String.format("%x", value));
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: Remove this line.

**Review by hangpark — CHANGES_REQUESTED** [review]

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package kr.ac.kaist.team888.hangulcharacter;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * HangulCharacter Loader.
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: Use JavaDoc reference.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * <p> Every HangulCharacter classes are loaded through this class.
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: Use JavaDoc reference.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+   * Getter of singleton instance.
+   *
+   * @return singleton instance
+   */
+  public static CharacterLoader getInstance() {
+    return Singleton.instance;
+  }
+
+  /**
+   * Get HangulCharacter class from Hangul enum class.
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: Use JavaDoc reference.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+   *
+   * @param hangul Hangul to get
+   * @return Correspond HangulCharacter class.
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: Use JavaDoc reference.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  private final String name;
+  private final int value;
+
+  Hangul(String name, int value) {
+    this.name = name;
+    this.value = value;
+  }
+
+  /**
+   * Get Hangul enum class from UNICODE value.
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: Use JavaDoc reference.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+   *
+   * @param value UNICODE value
+   * @return Correspond Hangul enum class
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: Use JavaDoc reference.

**Review by hangpark — CHANGES_REQUESTED** [review]
Use JavaDoc reference.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java (hunk 2 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java (hunk 3 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java (hunk 4 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java (hunk 5 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/CharacterLoader.java]: done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java (hunk 6 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java (hunk 7 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+  MEDI_WAE       ("Wae", 0x116D),           // ㅙ
+  MEDI_OE        ("Oe",  0x116E),           // ㅚ
+  MEDI_UH        ("Uh",  0x116F),           // ㅜ
+  MEDI_WO        ("Wo",  0x1170),           // ㅝ
+  MEDI_WE        ("We",  0x1171),           // ㅞ
+  MEDI_WI        ("Wi",  0x1172),           // ㅟ
+  MEDI_EU        ("Eu",  0x1173),           // ㅡ
+  MEDI_UI        ("Ui",  0x1174),           // ㅢ
+  MEDI_IH        ("Ih",  0x1175),           // ㅣ
+  FIN_GIYOUK        ("", 0x11A8),
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: `FIN_GIYEOK`

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+  FIN_SS_GIYOUK     ("", 0x11A9),
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: `FIN_SS_GIYEOK`

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+  FIN_GIYOUK_SIOT   ("", 0x11AA),
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: `FIN_GIYEOK_SIOT`

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+  FIN_NIEUN         ("", 0x11AB),
+  FIN_NIEUN_JIEUT   ("", 0x11AC),
+  FIN_NIEUN_HIEUT   ("", 0x11AD),
+  FIN_DIGEUT        ("", 0x11AE),
+  FIN_RIEUL         ("", 0x11AF),
+  FIN_RIEUL_GIYOUK  ("", 0x11B0),
```
Comment (hangpark) [inline app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: `FIN_RIEUL_GIYEOK`

**Review by hangpark — CHANGES_REQUESTED** [review]
Change `GIYOUK` to `GIYEOK`.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java (hunk 8 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: Done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java (hunk 9 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: Done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java (hunk 10 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: Done.

File: app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java (hunk 11 above)
Comment (xeoh) [reply app/src/main/java/kr/ac/kaist/team888/hangulcharacter/Hangul.java]: Done.

**Review by hangpark — APPROVED** [review]

**Pull request merged**